libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -501,10 +816,35 @@ void WaveClip::AppendLegacySharedBlock(
 {
    // This is a special use function for legacy files only and this assertion
    // does not need to be relaxed.  The clip is in a still unzipped track.
-   assert(GetWidth() == 1);
+   assert(NChannels() == 1);
    mSequences[0]->AppendSharedBlock( pBlock );
 }
 
+bool WaveClip::Append(size_t iChannel, const size_t nChannels,
+   constSamplePtr buffers[], sampleFormat format,
+   size_t len, unsigned int stride, sampleFormat effectiveFormat)
+{
+   assert(iChannel < NChannels());
+   assert(iChannel + nChannels <= NChannels());
+
+   // No requirement or promise of the strong invariant, and therefore no
+   // need for Transaction
+
+   //wxLogDebug(wxT("Append: len=%lli"), (long long) len);
```

Comment:
Intentionally so.

See #6107 which proposes a fix for the slow importing in the master branch.  That fix will conflict too much with wave-clip-refactoring, so if it is merged, it should be reverted before wave-clip-refactoring is merged to master.

I need an alternative fix for performance, and that is to remove the Transaction.  In fact I suggest #6107 also just remove it.

There is strong exception safety for operations on individual Sequence objects, meaning in case of exception, the previous state of the objects is restored unchanged.

There is only weak exception safety for WaveClip objects operating on multiple Sequences, unless there is also a Transaction, to handle the possibility that one Sequence is updated, but an exception happens during the update of the other.  Each Sequence remains self-consistent, but the WaveClip might not achieve strong exception safety, or the "strong invariant" condition of equal length sequences.

Recall PR #6089 which you reviewed.  I decided I can't easily guarantee that consistency always.  It reduced the programming effort to allow Append to violate the strong condition temporarily.

Reply:
Ok. I didn't notice the comment line 830.